cdc/processor/pipeline/cyclic_mark.go

```diff
@@ -174,3 +181,39 @@ func extractReplicaID(markRow *model.RowChangedEvent) uint64 {
 	log.Panic("bad mark table, " + mark.CyclicReplicaIDCol + " not found")
 	return 0
 }
+
+// cyclicNodeContext implements the NodeContext, cyclicMarkNode can be reused in table actor
+// to buffer all messages with a queue, it will not block the actor system
+type cyclicNodeContext struct {
+	*actorNodeContext
+	queue list.List
+}
+
+func NewCyclicNodeContext(ctx *actorNodeContext) *cyclicNodeContext {
+	return &cyclicNodeContext{
+		actorNodeContext: ctx,
+	}
+}
+
+// SendToNextNode implement the NodeContext interface, push the message to a queue
+func (c *cyclicNodeContext) SendToNextNode(msg pipeline.Message) {
+	c.queue.PushBack(msg)
```

Comment:
Could you add a comment about memory usage? I'm afraid without limitation it could cause OOM in some extreme cases.

Reply:
we limit the queue size at TryHandleDataMessage